### PR TITLE
Remove regex

### PR DESCRIPTION
**Pull request by AshantiMutinta**
## PR summary
Removed regex backend
## testing/benchmarking notes

( if any manual testing or benchmarking was/should be done, add notes and/or screenshots here )

## followups

( any new tickets/concerns that were discovered or created during this work but aren't in scope for review here )

## changelog

- [x] if this is a code change that effects some consumer (e.g. zome developers) of holochain core,  then it has been added to [our between-release changelog](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md) with the format 

```markdown
- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)
```

## documentation

- [ ] this code has been documented according to our [docs checklist](https://hackmd.io/@freesig/Hk9AmKJNS)

### CODE REVIEW

File: crates/core_types/src/link/mod.rs [hunk 1]
```diff
@@ -94,6 +92,15 @@ impl<S: Into<String>> LinkMatch<S> {
     }
 }
 
+impl<S: Into<String>> Into<Option<String>> for LinkMatch<S> {
+    fn into(self) -> Option<String> {
+        match self {
+            LinkMatch::Exactly(s) => Some(s.into()),
+            _ => None,
```
Comment (neonphog) [inline crates/core_types/src/link/mod.rs]: Should we add the `Any` arm here?

File: crates/core_types/src/link/mod.rs [hunk 2]
```diff
@@ -84,7 +83,6 @@ impl<S: Into<String>> LinkMatch<S> {
         let re_string: String = match self {
             LinkMatch::Any => ".*".into(),
             LinkMatch::Exactly(s) => "^".to_owned() + &regex::escape(&s.into()) + "$",
```
Comment (neonphog) [inline crates/core_types/src/link/mod.rs]: do we still even need this function? - this is a pretty inefficient way to match a full string (similarly for the Any above).

File: crates/core_types/src/link/mod.rs (hunk 1 above)
Comment (AshantiMutinta) [reply crates/core_types/src/link/mod.rs]: Good catch! Added it

File: crates/core_types/src/link/mod.rs (hunk 2 above)
Comment (AshantiMutinta) [reply crates/core_types/src/link/mod.rs]: Yeah we don't need this anymore, wasnt using it anywhere

**Pull request merged**